crates/nu-protocol/src/value/primitive.rs

```diff
@@ -57,6 +57,16 @@ pub enum Primitive {
 }
 
 impl Primitive {
+    /// Utility function for determining if a primitive is a number type
+    /// Useful for functions that are only valid for number types like sum, min, max
+    pub fn is_number(&self) -> bool {
```

Comment:
Not using this currently, but something like this might be necessary if we want to address the bug that allows non-number types